Code/GraphMol/ChemReactions/ReactionUtils.cpp

```diff
@@ -172,19 +172,14 @@ bool isReactionTemplateMoleculeAgent(const ROMol &mol, double agentThreshold) {
 
 namespace {
 
-void getMappingNumAtomIdxMapReactants(
-    ChemicalReaction *rxn, std::map<int, Atom::ChiralType> &reactantMapping) {
-  for (MOL_SPTR_VECT::const_iterator reactIt = rxn->beginReactantTemplates();
-       reactIt != rxn->endReactantTemplates(); ++reactIt) {
-    for (ROMol::AtomIterator reactAtomIt = (*reactIt)->beginAtoms();
-         reactAtomIt != (*reactIt)->endAtoms(); ++reactAtomIt) {
-      if (!(*reactAtomIt)->hasProp("molAtomMapNumber")) {
-        continue;
-      }
+void getMappingNumAtomIdxMapReactants(ChemicalReaction *rxn, std::map<int,Atom::ChiralType>& reactantMapping){
+  for(MOL_SPTR_VECT::const_iterator reactIt=rxn->beginReactantTemplates();
```

Comment:
why not const ChemicalReaction & instead? (Actually I've never quite understood the need for PRECONDITIONing on null pointers)